src/components/ButtonBase/ButtonBase.style.ts

```diff
@@ -1,51 +1,51 @@
 import { css, SerializedStyles } from '@emotion/react';
-import { rem } from 'theme/utils';
 
-import { Theme } from '../../theme';
-import { getDisabled, getFocus, getHover, getPressed } from '../../theme/states';
+import stateTokens from '../../theme/states/states.tokens';
+import buttonTokens from '../Button/Button.tokens';
 import { ButtonBaseProps } from './ButtonBase';
 
-export const buttonBaseStyle =
-  ({ isBlock, sx }: Omit<ButtonBaseProps, 'buttonType' | 'ref'>) =>
-  (theme: Theme): SerializedStyles => {
-    /** @TODO replace all temporary css values with the new ones */
-
-    const backGroundColor = '#123456';
-    const borderWidth = 0;
+export const buttonWrapperStyle = ({
+  isBlock,
+}: Pick<ButtonBaseProps, 'isBlock'>): SerializedStyles =>
+  css({ position: 'relative', width: isBlock ? '100%' : 'fit-content' });
 
+export const buttonBaseStyle =
+  ({
+    type = 'primary',
+    isBlock,
+    isLoading,
+    isDisabled,
+    sx,
+  }: Omit<ButtonBaseProps, 'buttonType' | 'ref'>) =>
+  (): SerializedStyles => {
     const baseButtonStyles = {
-      fontSize: rem(14),
-      fontWeight: theme.globals.typography.weights.get('medium'),
-      color: 'white',
+      color: buttonTokens.color[type].textColor,
       width: isBlock ? '100%' : undefined,
-      backgroundColor: backGroundColor,
-      padding: `0 ${theme.globals.spacing.get('6')}`,
-      height: rem(36),
-      borderRadius: theme.globals.spacing.get('3'),
-      border: `solid ${rem(borderWidth)} ${backGroundColor}`,
+      backgroundColor: stateTokens[isLoading ? 'active' : 'inactive'].backgroundColor[type],
+      padding: `${buttonTokens.spacing.textButton.paddingVertical} ${buttonTokens.spacing.textButton.paddingHorizontal}`,
+      borderRadius: buttonTokens.borderRadius.text,
+      border: `solid ${buttonTokens.borderWidth[1]} ${buttonTokens.color[type].borderColor}`,
       cursor: 'pointer',
       transition: 'background-color,border 150ms linear',
-      ':focus-visible:not(:disabled)': {
-        outline: getFocus({ theme, borderWidth: borderWidth }).styleOutline,
-      },
+
+      ':focus-visible:not(:disabled)': stateTokens.focus,
```

Comment:
also here you miss the `getFocus` functionality where you passed the `theme` in order to pick `semantic` as the theme from `theme.tokens`